test/unity/toolkits/activity_classifier/test_activity_classifier.cxx

```diff
@@ -0,0 +1,304 @@
+
+  // test for classify()
+  // test for per_window
+  gl_sframe predict_result = classifier.classify(gl_sframe(), "per_window");
+
+  // some simple tests for shape, column names
+  TS_ASSERT_EQUALS(predict_result.size(), session_id.size());
+  std::vector<std::string> column_names = predict_result.column_names();
+  TS_ASSERT_EQUALS(column_names.size(), 4);
+  TS_ASSERT_EQUALS(column_names[0], "prediction_id");
+  TS_ASSERT_EQUALS(column_names[1], "exp_id");
+  TS_ASSERT_EQUALS(column_names[2], "class");
+  TS_ASSERT_EQUALS(column_names[3], "probability");
+
+  // test prediction_id
+  gl_sarray prediction_id_array = predict_result["prediction_id"];
+  for (size_t i = 0; i < prediction_id_array.size(); i++) {
+    TS_ASSERT_EQUALS(prediction_id_array[i], i);
+  }
+
+  // test exp_id
+  gl_sarray exp_id_array = predict_result["exp_id"];
+  for (size_t i = 0; i < exp_id_array.size(); i++) {
+    TS_ASSERT_EQUALS(exp_id_array[i], session_id[i]);
+  }
+
+  // build the ground truth class and prob
+  std::vector<float> gt_prob;
+  std::vector<flexible_type> gt_class;
+  for (size_t i = 0; i < predict_probability.size(); i++) {
+    std::vector<flex_float> x = predict_probability[i];
+    auto max_it = std::max_element(x.begin(), x.end());
+    gt_prob.push_back(x[max_it - x.begin()]);
+    gt_class.push_back(class_labels[max_it - x.begin()]);
+  }
+
+  // test class
+  gl_sarray class_array = predict_result["class"];
+  for (size_t i = 0; i < gt_class.size(); i++) {
+    TS_ASSERT_EQUALS(gt_class[i], class_array[i]);
+  }
+
+  // test prob
+  gl_sarray prob_array = predict_result["probability"];
+  for (size_t i = 0; i < gt_prob.size(); i++) {
+    TS_ASSERT_EQUALS((float)gt_prob[i], (float)prob_array[i]);
+  }
+
+  // test for per_row
+  predict_result = classifier.classify(gl_sframe(), "per_row");
+
+  // simple test for shape, column names
+  TS_ASSERT_EQUALS(predict_result.size(), test_num_examples);
+  column_names = predict_result.column_names();
+  TS_ASSERT_EQUALS(column_names.size(), 2);
+  TS_ASSERT_EQUALS(column_names[0], "class");
+  TS_ASSERT_EQUALS(column_names[1], "probability");
+
+  // ground truth class and probability for per_row
+  std::vector<float> gt_prob_per_row;
+  std::vector<std::string> gt_class_per_row;
+  for (size_t i = 0; i < num_samples.size(); i++) {
+    for (size_t j = 0; j < num_samples[i]; j++) {
+      gt_prob_per_row.push_back(gt_prob[i]);
+      gt_class_per_row.push_back(gt_class[i]);
+    }
+  }
+
+  // test class and prob
+  gl_sarray prob_array_per_row = predict_result["probability"];
+  gl_sarray class_array_per_row = predict_result["class"];
+  TS_ASSERT_EQUALS(gt_prob_per_row.size(), predict_result.size());
+  for (size_t i = 0; i < predict_result.size(); i++) {
+    TS_ASSERT_EQUALS((float)gt_prob_per_row[i], (float)prob_array_per_row[i]);
+    TS_ASSERT_EQUALS(gt_class_per_row[i], class_array_per_row[i]);
+  }
+
+  // test for predict()
+  // ouput_type = class
```

Comment:
My 2c: It might be helpful to a reader, if the comments explain the test case a bit more. Such as, summarize the two lines as:
Test predict() using parameter `output_type` = 'class' 

Similarly for the comments below

Reply:
okay! Thanks!!